gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableMetadataValidatorUtils.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.data.management.copy.iceberg;
+
+import java.io.IOException;
+
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableMetadata;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Validator for Iceberg table metadata, ensuring that the given tables metadata have same schema and partition spec.
+ */
+@Slf4j
+public class IcebergTableMetadataValidatorUtils {
+
+  private IcebergTableMetadataValidatorUtils() {
+    // Do not instantiate
+  }
+
+  /**
+   * Compares the metadata of the given two iceberg tables.
+   * <ul>
+   *   <li>First compares the schema of the metadata.</li>
+   *   <li>Then compares the partition spec of the metadata.</li>
+   * </ul>
+   * @param tableAMetadata  the metadata of the first table
+   * @param tableBMetadata the metadata of the second table
+   * @param strictEqualityForPartitionSpec boolean value to control strictness of partition spec comparison
+   * @throws IOException if the schemas or partition spec do not match
+   */
+  public static void failUnlessCompatibleStructure(TableMetadata tableAMetadata,
+      TableMetadata tableBMetadata, boolean strictEqualityForPartitionSpec) throws IOException {
```

Comment:
sorry if I was the one to suggest this, but it `tableAMetadata` jumped out at me as a typo.  how about `tableMetadataA` or `tableMetadataX`?

(same w/ `schemaA`, below, etc.)